sw/otbn/code-snippets/mul256.S

```diff
@@ -0,0 +1,91 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/* 256-bit multiply example. Loads two 256-bit words from DMem into w1, w0.
+   Multiplies usings BN.MULQACC with the result placed into w3, w2 (upper half
+   in w3). */
+
+.section .text
+
+/* Load operands into WDRs */
+li x2, 0
+li x3, 1
+li x4, 32
+
+bn.lid x2, 0(x0)
+bn.lid x3, 0(x4)
+
+/* perform multiply, limbs are 64-bit and limb numbers refer to the lower of the
+   2 64-bit chunks output from a multiply e.g. the limb 3 section produces and
+   accumulates partial products for limbs 3 and 4 which overlaps with the limb 2
+   section (limbs 2 and 3) and the limb 4 section (limbs 4 and 5). */
```

Comment:
I found this comment quite hard to read. If you want to keep the naming in the comments below, maybe something like this would be clearer?
```
/* Perform multiplication. Limbs are 64-bit and the limb numbers in
   the comments refer to the lower of the two 64-bit chunks generated by a
   multiply. For example, the section marked "limb 3" produces and
   accumulates partial products for limbs 3 and 4. This overlaps with the "limb
   2" section (limbs 2 and 3) and the "limb 4" section (limbs 4 and 5). */
```
But it might be better to change the comments below to e.g. `limbs 0 and 1`, `limbs 1 and 2` etc. That would probably be easier to understand and might need less explanation.